packages/harpguru-core/src/components/callback-on-sourceglobalprops/hooks/use-setfrom-sourcecolumnbounds.ts

```diff
@@ -47,10 +52,23 @@ export const useSetFromSourceColumnBounds = (): void => {
     fullInteractionMatrix,
     nextSourceColumnBounds
   )
-  const nextLayoutFacts = reduceViewableMatrixToLayoutFacts(
+  const nextFullLayoutFacts = reduceMatrixToLayoutFacts(
+    prevLayoutFacts,
+    fullInteractionMatrix
+  )
+  const nextLayoutFacts = reduceMatrixToLayoutFacts(
     prevLayoutFacts,
     nextViewableInteractionMatrix
   )
+  const nextDynamicSizes = reduceLayoutFactsToDynamicSizes(
+    prevDynamicSizes,
+    nextLayoutFacts,
+    nextFullLayoutFacts
```

Comment:
Even better, because the order is important but these params are actually the same type, I've decided to pass them both within an object so that the params names are explicit and mistakes should be easier to spot.